fire/core.py

```diff
@@ -566,8 +569,13 @@ def _ParseFn(args):
       # If we're allowed *varargs or **kwargs, there's always capacity.
       capacity = True
 
-    if fn_keywords is None and kwargs:
-      raise FireError('Unexpected kwargs present', kwargs)
+    extra_kw = set(kwargs) - set(fn_kwonly)
+    if fn_keywords is None and extra_kw:
+      raise FireError('Unexpected kwargs present:', extra_kw)
+
+    missing_kwonly = set(required_kwonly) - set(kwargs)
+    if missing_kwonly:
+      raise FireError('Missing required flags:', missing_kwonly)
```

Comment:
That's a fair point.
Yeah, currently this error does surface to the user, so your string sgtm.
There's definitely room to clean up the Fire trace output, so we can think about changing how these errors are displayed if/when we go down that route.